### ISSUE TITLE
Dollar Amount is Read Off Incorrectly When Using TalkBack feature

### ISSUE THREAD

**Issue by JB184351** [woocommerce/woocommerce-android#7408]
**Describe the bug**
When adding an existing product to order the price isn’t read off correctly when using TalkBack, it is read as Dollar, amount, dot zero zero, obviously this isn’t the correct way to read this, but haven’t seen too many instances of this in the rest of the app so I’m guessing this could be fixed here.

**To Reproduce**

1. Go to the 'Order' tab
3. Scroll down to 'Create a New Note'
4. Toggle On TalkBack
5. Select the Product
6. Notice how the dollar amount is read back when using TalkBack

**Screenshots**
<img src="https://user-images.githubusercontent.com/36495451/190729019-bf10a4a6-17b5-44b6-919e-f2e7eaccecdc.jpg" width="300"/>

**Expected behavior**
The Dollar Amount should be read back clearly.

**Isolating the problem (mark completed items with an [x]):**
- [X] I have deactivated other plugins and confirmed this bug occurs when only WooCommerce plugin is active.
- [X] This bug happens with a default WordPress theme active, or [Storefront](https://woocommerce.com/storefront/).
- [X] I can reproduce this bug consistently using the steps above.

**Mobile Environment**
Please include:
- Device: Samsung Galaxy A51
- Android version: 12
- WooCommerce Android version: 10.1
